### PR TITLE
feat: add search query from URL

### PR DESCRIPTION
**Pull request by spectre007**

### CODE REVIEW

File: extra/js/URLSearchParams.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+function getURLParams() {
+    return new Proxy(new URLSearchParams(window.location.search), {
+        get: (searchParams, prop) => searchParams.get(prop),
+    });
+}
+
+// from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent
+function containsEncodedComponents(x) {
```
Comment (timurbazhirov) [inline extra/js/URLSearchParams.js:8]: Let's use a JSDoc docstring pls

File: extra/js/URLSearchParams.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+function getURLParams() {
```
Comment (timurbazhirov) [inline extra/js/URLSearchParams.js:1]: We've been reserving the CapitalName convention for React components, maybe `url_parameters.js`?

File: extra/js/URLSearchParams.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    });
+}
+
+// from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent
+function containsEncodedComponents(x) {
+  // ie ?,=,&,/ etc
+  return (decodeURI(x) !== decodeURIComponent(x));
+}
+
+function injectSearchfromURL(searchToken = "searchText") {
```
Comment (timurbazhirov) [inline extra/js/URLSearchParams.js:13]: is search token a URL decoded string? Maybe let's add a docstring

Comment (spectre007) [reply extra/js/URLSearchParams.js:13]: `searchToken = "searchText"` just defines the query key, for instance, `docs.mat3ra.com/?searchText=Property`. The keys and values are automatically grabbed by `URLSearchParams()`.

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**